File: app/src/main/java/awais/instagrabber/activities/BaseLanguageActivity.java
package awais.instagrabber.activities;

import android.os.Bundle;

import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;

import awais.instagrabber.utils.LocaleUtils;
import awais.instagrabber.utils.Utils;

public abstract class BaseLanguageActivity extends AppCompatActivity {
    protected BaseLanguageActivity() {
        LocaleUtils.updateConfig(this);
    }

    @Override
    protected void onCreate(@Nullable final Bundle savedInstanceState) {
        Utils.changeTheme(this);
        super.onCreate(savedInstanceState);
    }
}

File: app/src/main/java/awais/instagrabber/activities/BaseLanguageActivity.kt
package awais.instagrabber.activities

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import awais.instagrabber.utils.LocaleUtils
import awais.instagrabber.utils.ThemeUtils

abstract class BaseLanguageActivity protected constructor() : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        ThemeUtils.changeTheme(this)
        super.onCreate(savedInstanceState)
    }

    init {
        @Suppress("LeakingThis")
        LocaleUtils.updateConfig(this)
    }
}